From 82743f2a4d8fc099b17cac7d3b8f83ea13328916 Mon Sep 17 00:00:00 2001
From: Antoine Belvire <antoine.belvire@laposte.net>
Date: Wed, 24 Jun 2015 20:07:19 +0200
Subject: [PATCH] restrict is a reserved keyword in C99

restrict is a reserved keyword in C99/C11, therefore it can't be used as a variable name.
This commit fix compilation with GCC 5 (which now uses -std=gnu11 as default mode instead of -std=gnu89).

Signed-off-by: Antoine Belvire <antoine.belvire@laposte.net>
Signed-off-by: Eric B Munson <emunson@mgebm.net>
---
 hugeutils.c | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/hugeutils.c b/hugeutils.c
index 53a7fbd..60488e8 100644
--- a/hugeutils.c
+++ b/hugeutils.c
@@ -301,14 +301,14 @@ void hugetlbfs_setup_env()
 
 	env = getenv("HUGETLB_RESTRICT_EXE");
 	if (env) {
-		char *p, *tok, *exe, buf[MAX_EXE+1], restrict[MAX_EXE];
+		char *p, *tok, *exe, buf[MAX_EXE+1], restriction[MAX_EXE];
 		int found = 0;
 
 		exe = get_exe_name(buf, sizeof buf);
 		DEBUG("Found HUGETLB_RESTRICT_EXE, this exe is \"%s\"\n", exe);
-		strncpy(restrict, env, sizeof restrict);
-		restrict[sizeof(restrict)-1] = 0;
-		for (p = restrict; (tok = strtok(p, ":")) != NULL; p = NULL) {
+		strncpy(restriction, env, sizeof restriction);
+		restriction[sizeof(restriction)-1] = 0;
+		for (p = restriction; (tok = strtok(p, ":")) != NULL; p = NULL) {
 			DEBUG("  ...check exe match for \"%s\"\n",  tok);
 			if (strcmp(tok, exe) == 0) {
 				found = 1;
-- 
1.8.3.1

